cylc/flow/pathutil.py

```diff
@@ -213,10 +232,6 @@ def make_symlink(path: Union[Path, str], target: Union[Path, str]) -> bool:
             raise WorkflowFilesError(
                 f"Error when symlinking. Failed to unlink bad symlink {path}.")
     target.mkdir(parents=True, exist_ok=True)
-    if path.exists():
-        # Trying to link to itself; no symlink needed
-        # (e.g. path's parent is symlink to target's parent)
-        return False
```

Comment:
Line 226 has the the return false.

Reply:
But this checks if the path exists _after_ the target is created.
1. When the `~/cylc-run/foo/share -> ~/sym-share/cylc-run/foo/share` symlink & target already exist, and we're now on `share/cycle`
2. The line `target.mkdir(...)` above will create `~/sym-share/cylc-run/foo/share/cycle`.
3. At this point `path` exists now - `path` is `~/cylc-run/foo/share/cycle`, but because `~/cylc-run/foo/share` is a symlink, `path` is actually  `~/sym-share/cylc-run/foo/share/cycle`

(Sorry, the comment in the code doesn't give a good explanation I now realise! And I should have created a test case for `share` and `share/cycle` being the same)